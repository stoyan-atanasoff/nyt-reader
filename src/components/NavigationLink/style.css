a.drawer-link {
  text-decoration: none;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 16px;
  color: #db133e;
  padding: 13px 30px;
  display: block;
  margin: 1px -19px;
}
a.drawer-link.selected {
  background: #db133e;
  color: #fff;
}
a.drawer-link:active {
  background: #41addd;
  color: #fff;
}
a.drawer-link:hover {
  background: #db133e;
  color: #fff;
}

a.header-link {
  text-decoration: none;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 16px;
  color: #fff;
  padding: 13px 30px;
  display: inline-block;
}
a.header-link.selected {
  background: #db133e;
  color: #fff;
}
a.header-link:active {
  background: #41addd;
  color: #fff;
}
a.header-link:hover {
  background: #db133e;
  color: #fff;
}
