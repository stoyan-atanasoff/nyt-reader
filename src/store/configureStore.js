import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sagas from '../sagas';
import createReducer from '../reducers';
import {routerMiddleware} from "connected-react-router";

export const history = createBrowserHistory();

export default function (initialState = {}) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  // if (typeof window === 'object') {
  //   if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
  //     composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({});
  // }
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const middleWares = [sagaMiddleware, routerMiddleware(history)];
  const enhancers = [applyMiddleware(...middleWares)];
  const store = createStore(createReducer(), initialState, composeEnhancers(...enhancers));
  sagaMiddleware.run(sagas);

  return store;
}
