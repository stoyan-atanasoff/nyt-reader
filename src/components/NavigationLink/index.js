import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

const NavigationLink = (props) => <Link {...props}>{props.children}</Link>;
export default NavigationLink;
