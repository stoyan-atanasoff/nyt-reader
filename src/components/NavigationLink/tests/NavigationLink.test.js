import React from 'react';
import NavigationLink from '../index';
import { Router } from 'react-router-dom';
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {history} from "../../../store/configureStore";
import configureStore from "redux-mock-store";

const initialState = {};
const mockStore = configureStore();

describe('NavigationLink', () => {
  it('Navigation link should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={mockStore(initialState)}>
        <Router history={history}>
          <NavigationLink to={'/'}/>
        </Router>
      </Provider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
