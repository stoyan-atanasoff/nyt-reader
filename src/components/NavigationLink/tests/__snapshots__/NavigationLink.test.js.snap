// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLink Navigation link should render and match the snapshot 1`] = `
<a
  href="/"
/>
`;
